File: java/trees/Tree.java
import java.util.*;

public class Tree<E> {
    private Node<E> root;
    private int size;

    public Tree() { root=null; size=0; }
    
    public Tree(E s) {
        root = new Node<E>(s);
        size=1;
    }

    public boolean isEmpty() { return root==null;}

    public void setRoot(Node<E> root) {
        this.root = root;
    }

    public Node<E> getRoot() { return root; }

    public int size() {
        this.size = 0;
        if(!isEmpty()) { 
            this.size+=getChildrenSize(root);
        }
        return this.size;
    }

    private int getChildrenSize(Node<E> node) {
        List<Node<E>> children = new ArrayList<Node<E>>();
        children = node.getChildren();
        if (children.size()!=0) {
            for (Node<E> child : children) {
                this.size+=getChildrenSize(child);
            }
        }
        return 1;
    }
}

File: python/data-structures/tree.py
from tnode import TNode
class Tree:
    def __init__(self):
        self.root = None
        self.size = 0
    
    def get_root(self):
        return self.root
    
    def size(self):
        return self.size
    
    def set_root(self,node):
        self.root = TNode(node)
        self.size+=1

    def __str__(self):
        return str(self.root)


# Tree test
# t= Tree()
# t.set_root("1")
# root = t.get_root()
# m1 = TNode("2")
# m2 = TNode("3")
# m3 = TNode("4")
# root.add_children(TNode(m1,parent=root))
# root.add_children(TNode(m2,parent=root))
# root.add_children(TNode(m3,parent=root))